dask_cuda/benchmarks/local_cupy.py

```diff
@@ -210,63 +212,72 @@ async def run(args):
-            print(f"Protocol           | {args.protocol}")
-            print(f"Device(s)          | {args.devs}")
+            print_separator(separator="-")
+            print_key_value(key="Operation", value=f"{args.operation}")
+            print_key_value(key="User size", value=f"{args.size}")
+            print_key_value(key="User second size", value=f"{args.second_size}")
+            print_key_value(key="User chunk size", value=f"{args.size}")
+            print_key_value(key="Compute shape", value=f"{size}")
+            print_key_value(key="Compute chunk size", value=f"{chunksize}")
+            print_key_value(
+                key="Ignore size", value=f"{format_bytes(args.ignore_size)}"
+            )
+            print_key_value(key="Device(s)", value=f"{args.devs}")
             if args.device_memory_limit:
-                print(f"Memory limit       | {format_bytes(args.device_memory_limit)}")
-            print(f"Worker Thread(s)   | {args.threads_per_worker}")
-            print("==========================")
-            print("Wall-clock         | npartitions")
-            print("--------------------------")
-            for (took, npartitions) in took_list:
-                t = format_time(took)
-                t += " " * (11 - len(t))
-                print(f"{t}        | {npartitions}")
-            print("==========================")
-            print("(w1,w2)            | 25% 50% 75% (total nbytes)")
-            print("--------------------------")
+                print_key_value(
+                    key="Device memory limit",
+                    value=f"{format_bytes(args.device_memory_limit)}",
+                )
+            print_key_value(key="RMM Pool", value=f"{not args.disable_rmm_pool}")
+            print_key_value(key="Protocol", value=f"{args.protocol}")
+            if args.protocol == "ucx":
+                print_key_value(key="TCP", value=f"{args.enable_tcp_over_ucx}")
+                print_key_value(key="InfiniBand", value=f"{args.enable_infiniband}")
+                print_key_value(key="NVLink", value=f"{args.enable_nvlink}")
+            print_key_value(key="Worker thread(s)", value=f"{args.threads_per_worker}")
+            print_separator(separator="=")
+            print_key_value(key="Wall clock", value="Partitions")
+            print_separator(separator="-")
+            times = []
+            for idx, (took, npartitions) in enumerate(took_list):
+                m = format_time(took)
+                times.append(took)
+                print_key_value(key=f"{m}", value=npartitions)
+            times = np.asarray(times)
+            bandwidths_all = np.asarray(bandwidths_all)
+            print_separator(separator="=")
+            print_key_value(
+                key="Bandwidth",
+                value=f"{format_bytes(bandwidths_all.mean())}/s +/- "
+                f"{format_bytes(bandwidths_all.std())}/s",
+            )
```

Comment:
FWIW, and this is just the same as before, the arithmetic mean is generally the wrong thing for rate-like data, and the more correct option is the harmonic mean `len(xs)/sum(1/xs)`.

Reply:
Yeah, I was actually wondering if that was correct but didn't spend the time in investigating what would be the correct way to compute. Harmonic means do make sense indeed. However, it looks depressing 🙁 

```
Bandwidth                 | 10.18 GiB/s +/- 8.16 GiB/s
BandwidthH                | 3.47 GiB/s +/- 1.25 GiB/s
```

We _seemingly_ go down from 10.18GB/s to 3.47GB/s, of course the former is incorrect anyway, but it is sad.

I now replaced mean and standard deviation in https://github.com/rapidsai/dask-cuda/pull/938/commits/268f893b98faaccc87bea4f97060d502d513a15a, I am not sure whether standard deviation is correct, can you double-check?

Reply:
Standard deviations of inverse random variables is a huge can of worms :(, but as per [this stats.stackexchange post](https://stats.stackexchange.com/q/7471) there are a few options. Suppose we have our rates $X_i, \quad i=[1,\dots,n]$, then define the reciprocal weights $Y_i$ and calculate

$$
\begin{align}
\mu_Y &= \frac{1}{n}\sum_i Y_i \\
\text{Var}(Y) &= \frac{1}{n}\sum_i (Y_i - \mu_Y)^2
\end{align}
$$

Then the variance of the harmonic mean of $X_i$ ($H_X := \frac{1}{\mu_Y}$) is 

$$
\text{Var}_H = \frac{\text{Var}(Y)}{n \mu_Y^4}
$$

The above-linked post shows that this is a good estimate for uniformly distributed data at least.

So I think we can calculate harmonic mean and standard deviation as

```python
def hmean(data):
    return 1/np.mean(1/data)

def hstd(data):
    rmean = np.mean(1/data)
    rvar = np.var(1/data)
    return np.sqrt(rvar / (len(data) * rmean**4))
```

Reply:
Thanks, I saw that yesterday but wasn't totally sure that would be necessarily correct either. I've made the changes in https://github.com/rapidsai/dask-cuda/pull/938/commits/d120cd3e042382553b1b187a77abb6b08fe12c41 , thanks for the code as well!